etna/api/urls.py

```diff
@@ -50,6 +50,38 @@ def listing_view(self, request):
 
         self.check_query_parameters(queryset)
         queryset = self.filter_queryset(queryset)
+
+        if "include_aliases" not in request.GET:
+            alias_pages = queryset.filter(alias_of_id__isnull=False).values(
+                "id", "alias_of_id"
+            )
+            original_ids = set(
+                queryset.filter(alias_of_id__isnull=True).values_list("id", flat=True)
```

Comment:
Does this `queryset` have everything in it?

I'm wondering if we have filtered by an `offset` and `limit` (basically paginating the result) and the original isn't in this subset, we aren't getting the full list of IDs here.

Do we need to query a new queryset of ALL pages to ensure this is never the case?

Reply:
Pagination is applied after with `queryset = self.paginate_queryset(queryset)`, this filtering is applied after the main filtering is done (e.g. page type filtering, or `descendant_of` filtering) so that we're running queries on a smaller dataset - so this should be okay 🙂